packages/lesswrong/components/users/UsersProfile.jsx

```diff
@@ -153,10 +156,16 @@ class UsersProfile extends Component {
     
     if (!document || !document._id || document.deleted) {
       //eslint-disable-next-line no-console
-      console.error(`// missing user (_id/slug: ${documentId || slug})`);
+      console.error(`// missing user (_id/slug: ${slug})`);
       return <Components.Error404/>
     }
 
+    // Javascript redirect to make sure we are always on the most canonical URL for this user
+    if (slug !== document.slug) {
+      const canonicalUrl = Users.getProfileUrlFromSlug(document.slug);
+      router.replace(canonicalUrl);
```

Comment:
Add an early exit here, to avoid firing off graphql queries that will be immediately abandoned.
```suggestion
      router.replace(canonicalUrl);
      return null;
```